pkg/controller/installation/products/rhssouser/reconciler.go

```diff
@@ -0,0 +1,365 @@
+
+	return v1alpha1.PhaseCompleted, nil
+}
+
+func (r *Reconciler) handleProgressPhase(ctx context.Context, serverClient pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	kc := &aerogearv1.Keycloak{}
+	// if this errors, it can be ignored
+	err := serverClient.Get(ctx, pkgclient.ObjectKey{Name: keycloakName, Namespace: r.Config.GetNamespace()}, kc)
+	if err == nil && string(r.Config.GetProductVersion()) != kc.Status.Version {
+		r.Config.SetProductVersion(kc.Status.Version)
+		r.ConfigManager.WriteConfig(r.Config)
+	}
+
+	r.logger.Info("checking ready status for user-sso")
+	kcr := &aerogearv1.KeycloakRealm{}
+
+	err = serverClient.Get(ctx, pkgclient.ObjectKey{Name: keycloakRealmName, Namespace: r.Config.GetNamespace()}, kcr)
+	if err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to get keycloak realm custom resource")
+	}
+
+	if kcr.Status.Phase == aerogearv1.PhaseReconcile {
+		err = r.exportConfig(ctx, serverClient)
+		if err != nil {
+			return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to write user-sso config")
+		}
+
+		err = r.setupOpenshiftIDP(ctx, kcr, serverClient)
+		if err != nil {
+			return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to setup Openshift IDP for user-sso")
+		}
+
+		logrus.Infof("Keycloak has successfully processed the keycloakRealm for user-sso")
+		return v1alpha1.PhaseCompleted, nil
+	}
+
+	r.logger.Infof("user-sso KeycloakRealm status phase is: %s", kcr.Status.Phase)
+	return v1alpha1.PhaseInProgress, nil
+}
+
+func (r *Reconciler) exportConfig(ctx context.Context, serverClient pkgclient.Client) error {
+	kc := &aerogearv1.Keycloak{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      keycloakName,
+			Namespace: r.Config.GetNamespace(),
+		},
+	}
+	err := serverClient.Get(ctx, pkgclient.ObjectKey{Name: keycloakName, Namespace: r.Config.GetNamespace()}, kc)
+	if err != nil {
+		return pkgerr.Wrap(err, "could not retrieve keycloak custom resource for keycloak config for user-sso")
+	}
+	kcAdminCredSecretName := kc.Spec.AdminCredentials
+
+	kcAdminCredSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kcAdminCredSecretName,
+			Namespace: r.Config.GetNamespace(),
+		},
+	}
+	err = serverClient.Get(ctx, pkgclient.ObjectKey{Name: kcAdminCredSecretName, Namespace: r.Config.GetNamespace()}, kcAdminCredSecret)
+	if err != nil {
+		return pkgerr.Wrap(err, "could not retrieve keycloak admin credential secret for keycloak config for user-sso")
+	}
+	kcURLBytes := kcAdminCredSecret.Data["SSO_ADMIN_URL"]
+	r.Config.SetRealm(keycloakRealmName)
+	r.Config.SetHost(string(kcURLBytes))
+	err = r.ConfigManager.WriteConfig(r.Config)
+	if err != nil {
+		return pkgerr.Wrap(err, "could not update keycloak config for user-sso")
+	}
+	return nil
+}
+
+func (r *Reconciler) setupOpenshiftIDP(ctx context.Context, kcr *aerogearv1.KeycloakRealm, serverClient pkgclient.Client) error {
+	_, err := r.oauthv1Client.OAuthClients().Get(oauthId, metav1.GetOptions{})
```

Comment:
You should be able to use `r.ReconcileOauthClient` now 

https://github.com/integr8ly/integreatly-operator/pull/91/files#diff-329c149b9583e9a4fcdb670a2ba866f0R275-R289

Reply:
You can also use `r.ReconcileFinalizer` (just merged) to clean up oauth clients and any other cluster level resources created by your reocnciler